spec/formatting.md

```diff
@@ -19,6 +19,43 @@ the local variable takes precedence.
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
 
+## Bidirectional Isolation
+
+Formatted messages can be bidirectional,
+containing a mixture of left-to-right and right-to-left text.
+When replacing placeholders with formatted string values,
+the [Unicode Bidirectional Algorithm](http://www.unicode.org/reports/tr9/) [UAX9]
+can produce unexpected or undesirable effects, such as "spillover".
+In such cases, placeholders should be bidirectionally "isolated"
+so that the directionality of the text in the placeholder
+does not negatively affect the presentation of the overall string.
+
+Implementations may provide one or more strategies for isolation,
+but one such strategy must behave as follows,
+when formatting the message as a string:
+
+Considering a "formatted part" to be the formatted string representation of a Placeholder,
```

Comment:
I suggest we not use `part` to mean the stringified result of formatting. We've sort of established `formatToParts()` as the name of the method which returns structured formatting results, which implies that a _part_ is one such structured result of formatting. Perhaps _formatted elements_ or _formatted pattern elements_ would work better?